packages/odie-client/src/index.tsx

```diff
@@ -12,16 +13,38 @@ export const ODIE_THUMBS_DOWN_RATING_VALUE = 0;
 export const ODIE_THUMBS_UP_RATING_VALUE = 1;
 
 export const OdieAssistant: React.FC = () => {
-	const { chat, trackEvent, currentUser, isLoadingEnvironment, isLoadingExistingChat } =
-		useOdieAssistantContext();
+	const {
+		chat,
+		trackEvent,
+		currentUser,
+		isLoadingEnvironment,
+		isLoadingExistingChat,
+		isUserElegible,
+		isLoading,
+	} = useOdieAssistantContext();
 	const [ asisstantLoaded, setAssistantLoaded ] = useState( false );
 	const containerRef = useRef< HTMLDivElement >( null );
 	const messagesContainerRef = useRef< HTMLDivElement >( null );
+	const navigate = useNavigate();
 
 	useEffect( () => {
 		trackEvent( 'chatbox_view' );
 	}, [ trackEvent ] );
 
+	useEffect( () => {
+		const { pathname, search } = location;
+		const shouldRedirectHome = ! isUserElegible && ! isLoading;
+
+		// Prevent not eligible users from accessing wapuu.
+		if ( shouldRedirectHome ) {
+			trackEvent( 'calypso_helpcenter_redirect_not_eligible_user_to_homepage', {
+				pathname,
```

Comment:
These two values wont really help us. Pathname will always be /odie and search will always be empty. This is the virtual pathname inside the Help Center's MemoryRouter, not the global one. 

I would switch to window.location.pathname and search.